src/Http/Http.Abstractions/src/HttpProtocols.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Http
+{
+    /// <summary>
+    /// Contains methods to verify the request protocol of an HTTP request. 
+    /// </summary>
+    public static class HttpProtocols
+    {
+        // We are intentionally using 'static readonly' here instead of 'const'.
+        // 'const' values would be embedded into each assembly that used them
+        // and each consuming assembly would have a different 'string' instance.
+        // Using .'static readonly' means that all consumers get these exact same
+        // 'string' instance, which means the 'ReferenceEquals' checks below work
+        // and allow us to optimize comparisons when these constants are used.
+
+        // Please do NOT change these to 'const'
+        public static readonly string Http10 = "HTTP/1.0";
+        public static readonly string Http11 = "HTTP/1.1";
+        public static readonly string Http2 = "HTTP/2";
+        public static readonly string Http3 = "HTTP/3";
+
+        /// <summary>
+        /// Returns a value that indicates if the HTTP request protocol is HTTP/1.0. 
+        /// </summary>
+        /// <param name="protocol">The HTTP request protocol.</param>
+        /// <returns>
+        /// <see langword="true" /> if the protocol is HTTP/1.0; otherwise, <see langword="false" />.
+        /// </returns>
+        public static bool IsHttp10(string protocol)
+        {
+            return object.ReferenceEquals(Http10, protocol) || StringComparer.Ordinal.Equals(Http10, protocol);
```

Comment:
Should this check be case-sensitive or not? I found conflicting checks. NegotiateHandler is case insensitive and HttpProtocol.HttpVersion is case sensitive.

Reply:
Insensitive would be better. The reference equals should match most of the time regardless.

Reply:
I think case-sensitive is slightly better, since it's usually easier to make things less strict later rather than more strict. I don't think anyone is relying on case-insensitive protocol comparisons yet.

The argument that ReferenceEquals avoids any performance penalty is only true if the protocol actually is the first one that's checked. So for example, the new NegotiateHandler.IsSupportedProtocol getter would be slightly slower for protocols >= HTTP/2 assuming case insensitive comparison actually is actually slower.

Reply:
We discussed this in HTTP triage. Case-insensitive won out.